sqlglot/parser.py

```diff
@@ -3028,8 +3089,14 @@ def _parse_alter(self) -> t.Optional[exp.Expression]:
         this = self._parse_table(schema=True)
 
         actions: t.Optional[exp.Expression | t.List[t.Optional[exp.Expression]]] = None
-        if self._match_text_seq("ADD", advance=False):
-            actions = self._parse_csv(self._parse_add_column)
+
+        index = self._index
+        if self._match_text_seq("ADD"):
+            if self._match_set(self.ADD_CONSTRAINT_TOKENS):
+                actions = self._parse_csv(self._parse_add_constraint)
+            else:
+                self._retreat(index)
+                actions = self._parse_csv(self._parse_add_column)
```

Comment:
This retreat is useful because there are some dialects where you can have a csv list of `ADD COLUMN ....`, so `_parse_add_column` first matches that pair of tokens to enable the use of `_parse_csv`.